similarity_framework/src/impl/comparator/comparator_by_column.py

```diff
@@ -376,68 +344,127 @@ def compare(self, metadata1: DataFrameMetadata, metadata2: DataFrameMetadata, in
+            return self.__nonnumerical_compare1(metadata1, metadata2, index1, index2, column1_type, column2_type)
+
+        if column1_type == column2_type:
+            return 0
+        return 1
+
+
+class ComparatorByColumn(Comparator):
     """
     Comparator for comparing two tables
     """
 
+    @staticmethod
+    def from_settings(settings: AnalysisSettings) -> "ComparatorByColumn":
+        comparator = ComparatorByColumn()
+        if settings.size:
+            comparator.add_comparator_type(SizeHandler(settings.weights.size))
+        if settings.incomplete_columns:
+            comparator.add_comparator_type(IncompleteColumnsHandler(settings.weights.incomplete_columns))
+        if settings.exact_names:
+            comparator.add_comparator_type(ColumnExactNamesHandler(settings.weights.exact_names))
+        if settings.column_name_embeddings:
+            comparator.add_comparator_type(ColumnNamesEmbeddingsHandler(settings.weights.column_name_embeddings))
+        if settings.column_embeddings:
+            comparator.add_comparator_type(ColumnEmbeddingsHandler(settings.weights.column_embeddings))
+        if settings.kinds:
+            comparator.add_comparator_type(ColumnKindHandler(weight=settings.weights.kinds))
+        if settings.type_basic or settings.type_structural or settings.type_advanced:
+            comparator.add_comparator_type(ColumnTypeHandler(settings.weights.type))
+        if settings.distance_function:
+            func = HausdorffDistanceMin() if settings.distance_function == "HausdorffDistanceMin" else AverageDist()
+            comparator.set_distance_function(func)
+        return comparator
+
     def __init__(self):
         """
         Constructor for ComparatorByColumn
         """
-        self.comparator_type: list[ComparatorType] = []
-        self.table_comparators: list[TableComparator] = []
-        self.settings: set[Settings] = set()
-        self.distance_function = HausdorffDistanceMin()
+        super().__init__()
+        self.table_comparators: list[TableHandler] = []
 
-    def set_distance_function(self, distance_function: DistanceFunction) -> "ComparatorByColumn":
-        """
-        Set distance function for comparing two tables
-        """
-        self.distance_function = distance_function
-        return self
-
-    def set_settings(self, settings: set[Settings]) -> "ComparatorByColumn":
-        """
-        Set settings for comparing two tables
-        """
-        self.settings = settings
-        return self
-
-    def add_settings(self, setting: Settings) -> "ComparatorByColumn":
-        """
-        Add another setting for comparing two tables
-        """
-        self.settings.add(setting)
-        return self
-
-    def add_comparator_type(self, comparator: ComparatorType) -> "ComparatorByColumn":
+    def add_comparator_type(self, comparator: HandlerType) -> "ComparatorByColumn":
         """
         Add comparator
         """
```

Comment:
did you intend to finish this todo in this PR or in next one?

Reply:
no